### PR TITLE
Feature/025-part 2 : refactor code

### PR DESCRIPTION
**Pull request by SaXaPhonist**
- add lint rules
 - refactor code
 - add query params at courses endpoint

### CODE REVIEW

File: backend/src/controllers/auth/refresh.ts [hunk 1]
```diff
@@ -23,11 +20,11 @@ const refresh = async (req: Request, res: Response, next: TMiddlewareCall) => {
 
     const isValidToken = refreshToken === dbUser?.refreshToken;
     if (!isValidToken) {
-      throw new Error('access time expired');
+      throw new Error('refresh time expired');
     }
 
     const newTokens = await generateJWT(dbUser);
-    res.cookie('refreshToken', newTokens.refreshToken, {maxAge: TIME_30D_SEC, httpOnly: true});
+    res.cookie('refreshToken', newTokens.refreshToken, { maxAge: TIME_30D_SEC, httpOnly: true });
```
Comment (NikitaKhadnevich) [inline backend/src/controllers/auth/refresh.ts:27]: Dont need here this `secure: true` ?I mean after `httpOnly: true`.

File: frontend/src/pages/Profile/Profile.tsx [hunk 2]
```diff
@@ -48,15 +48,16 @@ const ProfileContent: React.FC<User> = ({
         </ListItem>
       </UserInfoList>
       <UserInfoList subheader={<UserListSubheader>Skills</UserListSubheader>}>
-        {skills?.map((skill, id) => (
-          <ListItem key={id}>
+        {skills?.map((skill) => (
+          <ListItem key={skill}>
             <UserListItem>{skill}</UserListItem>
```
Comment (NikitaKhadnevich) [inline frontend/src/pages/Profile/Profile.tsx:53]: Are `skills` uniq?

File: frontend/src/utils/helpers/getAuthResponseData.tsx [hunk 3]
```diff
@@ -6,4 +6,5 @@ export const getAuthResponseData = (): string | undefined => {
   if (backendData) {
     return JSON.parse(backendData);
   }
+  return undefined;
```
Comment (NikitaKhadnevich) [inline frontend/src/utils/helpers/getAuthResponseData.tsx:9]: For what `undefined`?

File: frontend/src/utils/helpers/getUserIdCookie.tsx [hunk 4]
```diff
@@ -6,4 +6,5 @@ export const getUserIdCookie = (): string | undefined => {
   if (backendData) {
     return JSON.parse(backendData);
   }
+  return undefined;
```
Comment (NikitaKhadnevich) [inline frontend/src/utils/helpers/getUserIdCookie.tsx:9]: !

File: backend/src/interfaces/Ientities/Icourses.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
-import { TMaterials } from './Imaterials';
+import { ObjectId } from 'mongoose';
 
 interface ICourse {
-  _id?: string;
+  _id?: ObjectId;
   title: string;
```
Comment (NikitaKhadnevich) [inline backend/src/interfaces/Ientities/Icourses.ts:5]: `_id`  also used on frontend interfaces and was `string`. Wont this fix crash out interfaces?

File: frontend/package.json [hunk 6]
```diff
@@ -10,6 +10,7 @@
     "@mui/icons-material": "^5.2.5",
     "@mui/material": "^5.2.4",
     "@mui/styles": "^5.2.3",
+    "@mui/system": "^5.2.8",
```
Comment (NikitaKhadnevich) [inline frontend/package.json:13]: We have duplicate on this line.

File: frontend/src/constants/authConstants.tsx [hunk 7]
```diff
@@ -1,16 +1,16 @@
...
+interface IRequestErrors {
   readonly [key: string]: string;
 }
-interface cookiesTypes {
+interface ICookiesTypes {
   readonly [key: string]: string;
 }
 
-const REQUEST_ERRORS: requestErrors = {
+const REQUEST_ERRORS: IRequestErrors = {
```
Comment (NikitaKhadnevich) [inline frontend/src/constants/authConstants.tsx:8]: We added this changes for unification with backend interfaces?

**Review by NikitaKhadnevich — CHANGES_REQUESTED** [review]
Have some questions. I suppose we will merge it after main `feature/v1-025`

File: frontend/src/utils/helpers/getUserIdCookie.tsx (hunk 4 above)
Comment (SaXaPhonist) [reply frontend/src/utils/helpers/getUserIdCookie.tsx:9]: https://eslint.org/docs/rules/consistent-return

File: backend/src/interfaces/Ientities/Icourses.ts (hunk 5 above)
Comment (SaXaPhonist) [reply backend/src/interfaces/Ientities/Icourses.ts:5]: it's string

**Review by NikitaKhadnevich — DISMISSED** [review]
UPDATE: Resolved changes and discussed fix.

**Review by Julia-Maliutsina — APPROVED** [review]

**Review by k-krasilnikova — APPROVED** [review]

**Pull request merged**